### PR TITLE
Add test coverage for rest-data-with-panache repositories

### PR DESCRIPTION
**Pull request by kshpak**
This work is part of https://github.com/quarkus-qe/quarkus-test-plans/blob/main/QUARKUS-976.md effort

### CODE REVIEW

File: 014-quarkus-panache-with-transactions-xa/src/main/java/io/quarkus/qe/rest/data/UserResource.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import io.quarkus.hibernate.orm.rest.data.panache.PanacheRepositoryResource;
+import io.quarkus.panache.common.Page;
+import io.quarkus.panache.common.Sort;
+import io.quarkus.rest.data.panache.MethodProperties;
+import io.quarkus.rest.data.panache.ResourceProperties;
+
+import java.util.List;
+
+@ResourceProperties(exposed = true, hal = true, halCollectionName = "user_list", paged = true, path = "users")
```
Comment (pjgg) [inline 014-quarkus-panache-with-transactions-xa/src/main/java/io/quarkus/qe/rest/data/UserResource.java:11]: exposed = true
paged = true

Are redundant annotation properties, by default are set to true, so could be removed

File: 014-quarkus-panache-with-transactions-xa/src/test/java/io/quarkus/qe/rest/data/UserResourceTest.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+import io.quarkus.test.junit.QuarkusTest;
+
+@QuarkusTest
+@QuarkusTestResource(MySqlDatabaseTestResource.class)
+class UserResourceTest {
+
+    private final static String NEW_USER_ID = "3";
+    @Test
+    void testAllRepositoryMethods() throws InterruptedException {
```
Comment (pjgg) [inline 014-quarkus-panache-with-transactions-xa/src/test/java/io/quarkus/qe/rest/data/UserResourceTest.java:22]: `InterruptedException` is never thrown

**Review by pjgg — CHANGES_REQUESTED** [review]

**Comment by pjgg** [comment]
There are just a few minor (cosmetic) changes, apart from that LGTM

**Review by Sgitario — COMMENTED** [review]
LGTM, but can you update the README to state why you created an additional entity? I guess that you wanted to have an entity without extending PanacheEntity and also have a resource when using a separated repository.

File: 014-quarkus-panache-with-transactions-xa/src/main/java/io/quarkus/qe/rest/data/UserResource.java (hunk 1 above)
Comment (kshpak) [reply 014-quarkus-panache-with-transactions-xa/src/main/java/io/quarkus/qe/rest/data/UserResource.java:11]: I did it on purpose, with the aim to see if those parameters may or may not change in the future, so then we aware of it

File: 014-quarkus-panache-with-transactions-xa/src/test/java/io/quarkus/qe/rest/data/UserResourceTest.java (hunk 2 above)
Comment (kshpak) [reply 014-quarkus-panache-with-transactions-xa/src/test/java/io/quarkus/qe/rest/data/UserResourceTest.java]: Resolved

**Comment by kshpak** [comment]
@pjgg @Sgitario comments were resolved. May you take another look pls.

**Review by Sgitario — APPROVED** [review]

**Review by pjgg — APPROVED** [review]

File: 014-quarkus-panache-with-transactions-xa/src/main/java/io/quarkus/qe/rest/data/UserResource.java (hunk 1 above)
Comment (kshpak) [reply 014-quarkus-panache-with-transactions-xa/src/main/java/io/quarkus/qe/rest/data/UserResource.java]: the defaults are better to be removed as it better to test whether they could be changed it the future

**Pull request merged**